discovery_engine_core/bindings/src/types/slice.rs

```diff
@@ -0,0 +1,72 @@
+// Copyright 2022 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+//! Modules containing FFI glue for `Box<[u8]>`.
+
+use std::{mem::MaybeUninit, slice};
+
+/// Allocates a slice of (uninitialized) bytes of given length.
+pub(super) fn alloc_uninitialized_slice<T>(len: usize) -> *mut T {
+    let mut vec = Vec::<MaybeUninit<T>>::with_capacity(len);
```

Comment:
Coming back here from my question on the slice on f32 since it applies to all slices.
So seems that the main difference is that here we don't have a type to name for `place` because we cannot write `* mut &[T]` that is what we want. 
What about creating our own struct 
```rust
struct Slice<T> {
  ptr: *mut T,
  len: usize,
}
```
To represent a slice a make the interface uniform with String and Vec?

Reply:
This is not a `* mut &[T]` it's a `Box<[T]>`.

It's also meant to be directly called by the type specific ffi-exposed methods, in which case we can't return a `Slice` but only `Box<Slice>` which would just add additional complexity with no value.

In difference to other constructors which create a type at a specific place, this is meant to create a place we pass to rust.

Reply:
But instead of having pointer and len separate we can have a struct that holds them together like a String and Vec will do. We can also implement this using the Vec directly without having our own struct.